### PR TITLE
Closes #5868: Correctly handle null title when migrating open tabs.

### LINKED ISSUE #5868: IllegalStateException: Expected a string but was NULL
https://sentry.prod.mozaws.net/operations/fenix-fennec-nightly/issues/7235232/

```
IllegalStateException: Expected a string but was NULL
    at android.util.JsonReader.nextString(JsonReader.java:408)
    at mozilla.components.support.migration.session.StreamingSessionStoreParser.parseWindows(StreamingSessionStoreParser.kt:28)
    at mozilla.components.support.migration.session.StreamingSessionStoreParser.parse$support_migration_release(StreamingSessionStoreParser.kt:7)
    at mozilla.components.support.migration.FennecSessionMigration.migrate(FennecSessionMigration.kt:12)
    at mozilla.components.support.migration.FennecMigrator.migrateOpenTabs(FennecMigrator.kt:14)
...
(6 additional frame(s) were not displayed)

[INFO] Expected a string but was NULL
```

### PR DESCRIPTION
**Pull request by pocmo**
Closes #5868.

### CODE REVIEW

File: components/support/migration/src/main/java/mozilla/components/support/migration/session/InMemorySessionStoreParser.kt [hunk 1]
```diff
@@ -44,7 +44,12 @@ internal object InMemorySessionStoreParser {
 
             val entry = entries.getJSONObject(index - 1)
             val url = entry.getString("url")
-            val title = entry.optString("title").ifEmpty { url }
+            val title = if (entry.isNull("title")) {
+                url
+            } else {
+                entry.optString("title").ifEmpty { url }
+            }
```
Comment (Amejia481) [inline components/support/migration/src/main/java/mozilla/components/support/migration/session/InMemorySessionStoreParser.kt]: nit: Can we do the same as we did below? `parsedTitle?.ifEmpty { url } ?: url`

Comment (pocmo) [reply components/support/migration/src/main/java/mozilla/components/support/migration/session/InMemorySessionStoreParser.kt]: `optString` unfortunately doesn't give us `null` values and will return an empty String if you ask it for a value that is `null`.

Comment (Amejia481) [reply components/support/migration/src/main/java/mozilla/components/support/migration/session/InMemorySessionStoreParser.kt]: Got it!

**Comment by pocmo** [comment]
bors r=Amejia481

**Pull request merged**